### ISSUE TITLE
Permit leading indentation on line following `\n` in `Layout/LineContinuationLeadingSpace`

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#13571]
## Is your feature request related to a problem? Please describe.

This string shows the preferred style for `Layout/LineContinuationLeadingSpace`, with the whitespace appearing at the end of the line.

```ruby
p 'This string is a really long line ' \
  'which reads better if the whitespace is trailing.'
```

This string shows "incorrect" leading whitespace:

```ruby
p "This string contains a multi-line list:\n" \
  "    - it reads better if the\n" \
  '    - indentation is leading'
```

Despite being "incorrect", this version is arguably easier to read, because it differs in the following ways:

- the previous lines each end in a newline (`"\n")`, and
- the leading whitespace is **indentation** for that line.

Currently, `Layout/LineContinuationLeadingSpace` auto-corrects to the following:

```ruby
p "This string contains a multi-line list:\n    " \
  "- it reads better if the\n    " \
  '- indentation is leading'
```

This version, with the indentation at the end of the previous line is arguably less readable.

The readability issue in cases like this stands in the way of some consumers adopting `Layout/LineContinuationLeadingSpace`.

## Describe the solution you'd like

It would be ideal if RuboCop treated this case differently and adapted the rules for `Layout/LineContinuationLeadingSpace` depending on the previous line:

- When whitespace appears without a newline (`"\n"`), continue enforcing **trailing** whitespace.
- When whitespace appears after a newline, enforce **leading** whitespace (indentation).

This would still provide a consistent and deterministic layout, while not sacrificing readability.

If desired, this could be considered an additional supported style, rather than the default.

```yaml
SupportedStyles:
  - leading
  - trailing
  - trailing_except_after_newline # << Proposed addition
```

## Describe alternatives you've considered

The example above is arguably best expressed as a `HEREDOC`

```ruby
p <<~HEREDOC.chomp
  This string contains a multi-line list:
      - it reads better if the
      - indentation is leading
HEREDOC
```

However, there are large codebases that already contain such strings expressed as line continuations, in which it would be infeasible to manually convert these to `HEREDOC`s.

## Additional context

[This Gist](https://gist.github.com/sambostock/c66609e22db070cff5438538a4f37cf2) has a working reproduction of the example above.

For reviewers, are there any edge cases I haven't considered in which it would be preferable to include whitespace on the same line as the newline?

**Comment by sambostock** [comment]
I've opened #13572 as a proof-of-concept which just changes the behavior for `EnforcedStyle: trailing`. If there is interest in this feature, I can flesh out the implementation.

**Comment by Earlopain** [comment]
I think I would be fine with this being the default. I don't particularly see why you'd want it the other way around and this doesn't introduce new offenses for code that was already formatted, right?

Definitly more reminiscent of heredocs though, the cop seems more intended for single spaces between words.

**Comment by sambostock** [comment]
> Definitly more reminiscent of heredocs though, the cop seems more intended for single spaces between words.

Yeah, I agree. I don't think the author anticipated the indentation edge case.

> this doesn't introduce new offenses for code that was already formatted, right?

This depends on if we want to _ignore_ these cases, or enforce a certain style.

I think we agree that

```ruby
p "List:\n" \
  '    - item'
```

is the most readably style and that we should allow a configuration that permits this. However, should we also allow

```ruby
p "List:\n    " \
  '- item'
```

or should we enforce that it is corrected to the other style?

----

Personally, I would like a config that enforces the first style for the indentation case, but enforces the trailing style for the standard single space case, but if we made that the default, then it would technically be a breaking change (unless we consider it a bug fix).

**Comment by sambostock** [comment]
Should we move forward with this?
